src/main/java/org/reaktivity/nukleus/kafka/internal/stream/KafkaClientMetaFactory.java

```diff
@@ -1332,6 +1332,15 @@ private void doEncodeRequest(
                 doNetworkData(traceId, budgetId, encodeBuffer, encodeOffset, encodeProgress);
             }
 
+            private void cancelNextRequestSignal()
+            {
+                if (nextRequestAt != NO_CANCEL_ID)
+                {
+                    signaler.cancel(nextRequestAt);
+                    nextRequestAt = NO_CANCEL_ID;
+                }
+            }
```

Comment:
Review all calls to `cancelNextRequestSignal()` including commented calls in this PR.